assets/js/components/setup/setup-wrapper.js

```diff
@@ -24,58 +24,45 @@ import { delay } from 'lodash';
 /**
  * WordPress dependencies
  */
-import { withFilters } from '@wordpress/components';
-import { Component, Fragment } from '@wordpress/element';
+import { Fragment } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
+import Data from 'googlesitekit-data';
 import Header from '../header';
 import Link from '../link';
 import HelpLink from '../help-link';
 import { getSiteKitAdminURL } from '../../util';
+import { STORE_NAME as CORE_SITE } from '../../googlesitekit/datastore/site/constants';
+import { STORE_NAME as CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
+const { useSelect } = Data;
 
-class BaseComponent extends Component {
-	render() {
-		const { children } = this.props;
-		return (
-			<Fragment>
-				{ children }
-			</Fragment>
-		);
-	}
-}
-
-class SetupWrapper extends Component {
-	constructor( props ) {
-		super( props );
+export default function SetupWrapper() {
+	const settingsPageURL = useSelect( ( select ) => select( CORE_SITE ).getAdminURL() );
+	const { moduleToSetup } = global._googlesitekitLegacyData.setup;
+	const module = useSelect( ( select ) => select( CORE_MODULES ).getModule( moduleToSetup ) );
 
-		const { moduleToSetup } = global._googlesitekitLegacyData.setup;
-		this.state = {
-			currentModule: moduleToSetup,
-		};
+	if ( ! module ) {
+		return null;
 	}
 
-	static loadSetupModule( slug ) {
-		// Disabled because this rule doesn't acknowledge our use of the variable
-		// as a component in JSX.
-		// eslint-disable-next-line @wordpress/no-unused-vars-before-return
-		const FilteredModuleSetup = withFilters( `googlesitekit.ModuleSetup-${ slug }` )( BaseComponent );
+	function loadSetupModule() {
+		const { setupComponent: SetupComponent } = module;
 
 		return (
-			<FilteredModuleSetup
-				finishSetup={ SetupWrapper.finishSetup }
-				onSettingsPage={ false }
-				isEditing={ true }
+			<SetupComponent
+				module={ module }
+				finishSetup={ finishSetup }
 			/>
 		);
 	}
 
 	/**
 	 * When module setup done, we redirect the user to Site Kit dashboard.
 	 */
-	static finishSetup() {
+	function finishSetup() {
```

Comment:
Update this function to be created with `useCallback` and `[]` dependencies to avoid unnecessary re-renders.